MIP/mip-draft-FFS/README.md

```diff
@@ -0,0 +1,214 @@
+
+Our objective is to enable transaction issuers to quickly get some guarantees that their transactions are correctly included in a block. The crypto-economic security is provided by a PoS protocol.
+
+The mechanism can be deployed independently for a chain, or used in combination with existing settlement mechanisms, such as ZK and optimistic settlements.
+
+As a result, users can rely and trust the **L2-finality**  to use as confirmation, or if the chain is configured to do so, wait for **L1-finality**, such as end of challenge window for fraud proofs (optimistic L2) or verification of a ZK-proof (validity L2).
+
+A introduction to FFS can be found in [this blog post on Fast-Finality Settlement](https://blog.movementlabs.xyz/article/security-and-fast-finality-settlement). A more detailed description of a (partial) implementation of the mechanism is available at [this blog post on Postconfirmations](https://blog.movementlabs.xyz/article/postconfirmations-L2s-rollups-blockchain-movement).
+
+This MIP provides an overview of an architecture of FFS, and its main components.
+
+## Specification
+
+### Overview
+
+The objective of FFS is to confirm that transactions are processed correctly. It does not relate to the ordering of transactions.
+
+At an abstract level, the blockchain increases by a new L2-block in each (L2) round, and this L2-block is the successor of the L2-block in the previous round, the _predecessor_. Initially, there is a _genesis_ block with no predecessor.
+
+**Batch**. Each round corresponds to the processing of a _batch_ of transactions which is proposed by the _sequencer_ (can be centralised, decentralised, shared). 
+
+**Block**. A node with execution capability is then in charge of validating the transactions in the batch and calculate the new state. Since the batch is given by the sequencer, the new state and the state roots for a block are deterministic. For a batch $b$ the state is $S_b$ and the state root is $H(S_b)$. From the batch $b$ and the state $S_b$ the block $B$ is computed (which contains the information of the batch and the state root). 
+
+**Local validation**. Since the block is deterministically calculated we say a block (and the associated new state) is _validated locally_ once the execution engine calculates it from the batch. 
+
+**L2-Confirmation**. FFS aims to _confirm_ the validity of each produced block, in each round. The validity judgement to be made is: 
+> [!NOTE]
+> Given a block $B$ (predecessor), a batch of transactions $txs$ and a successor block $B'$, is $B'$ the^[the MoveVM is deterministic and there can be only valid successor.] _correct_ successor of $B$ after executing the sequence of transactions $txs$?
+
+The term _correct_ means that the successor block $B'$ (and the state it represents) has been computed in accordance with the semantics of the MoveVM, which we denote  $B \xrightarrow{\ txs \ } B'$.
```

Comment:
MoveVM is a great choice but I am presuming we could use any VM or STF here, are we better to generalise here as we have in the architecture as a whole?

Reply:
yes. Any VM (deterministic) or reference semantics can be used.
We could write (e.g. the MoveVM)?